src/drivers/xpad_uhid/hid_report.rs

```diff
@@ -0,0 +1,251 @@
+//E: 000000.228948 16 0b 00 80 00 80 00 00 00 80 00 00 00 00 00 00 00
+
+//RIGHTSTICK_RIGHT
+//# ReportID: 11 / X:  32768 | Y:  32768
+//#              | Rx:  64512 | Ry:  33792 | Z:     0 | # | Rz:     0 | # | Button: 0  0  0  0  0  0  0  0  0  0  0  0  0  0 | # | Hat switch:   0 | #
+//E: 000000.732994 16 0b 00 80 00 80 00 fc 00 84 00 00 00 00 00 00 00
+
+/// DInput Data Report
+#[derive(PackedStruct, Debug, Copy, Clone, PartialEq)]
+#[packed_struct(bit_numbering = "msb0", size_bytes = "16")]
+pub struct DInputDataReport {
+    // BYTE 0
+    #[packed_field(bytes = "0")]
+    pub report_id: u8,
+
+    // Axes
+    // BYTES 1-2
+    #[packed_field(bytes = "1..=2", endian = "lsb")]
+    pub l_stick_x: u16,
+    // BYTES 3-4
+    #[packed_field(bytes = "3..=4", endian = "lsb")]
+    pub l_stick_y: u16,
+    // BYTES 5-6
+    #[packed_field(bytes = "5..=6", endian = "lsb")]
+    pub r_stick_x: u16,
+    // BYTES 7-8
+    #[packed_field(bytes = "7..=8", endian = "lsb")]
+    pub r_stick_y: u16,
+    // BYTES 9-10
+    #[packed_field(bytes = "9..=10", endian = "lsb")]
+    pub trigger_l: u16,
+    // BYTES 11-12
+    #[packed_field(bytes = "11..=12", endian = "lsb")]
+    pub trigger_r: u16,
+
+    // Buttons
+    //BYTE 13
+    #[packed_field(bits = "104")]
+    pub menu: bool,
+    #[packed_field(bits = "105")]
+    pub view: bool,
+    #[packed_field(bits = "106")]
+    pub rb: bool,
+    #[packed_field(bits = "107")]
+    pub lb: bool,
+    #[packed_field(bits = "108")]
+    pub y: bool,
+    #[packed_field(bits = "109")]
+    pub x: bool,
+    #[packed_field(bits = "110")]
+    pub b: bool,
+    #[packed_field(bits = "111")]
+    pub a: bool,
+    //BYTE 14
+    #[packed_field(bits = "118")]
+    pub thumb_r: bool,
+    #[packed_field(bits = "119")]
+    pub thumb_l: bool,
+    //BYTE 15
+    #[packed_field(bytes = "15")]
+    pub dpad_state: u8,
+}
+
+impl Default for DInputDataReport {
+    fn default() -> Self {
+        Self {
+            report_id: Default::default(),
+            l_stick_x: Default::default(),
+            l_stick_y: Default::default(),
+            r_stick_x: Default::default(),
+            r_stick_y: Default::default(),
+            trigger_l: Default::default(),
+            trigger_r: Default::default(),
+            a: Default::default(),
+            b: Default::default(),
+            x: Default::default(),
+            y: Default::default(),
+            rb: Default::default(),
+            lb: Default::default(),
+            view: Default::default(),
+            menu: Default::default(),
+            thumb_l: Default::default(),
+            thumb_r: Default::default(),
+            dpad_state: Default::default(),
+        }
+    }
+}
+/// State data can be emitted from Output events to change data such as rumble.
+#[derive(PackedStruct, Debug, Copy, Clone, PartialEq, Default)]
+#[packed_struct(bit_numbering = "msb0", size_bytes = "47")]
+pub struct XpadUhidOutputData {}
```

Comment:
What data should be here?

Reply:
Don't know yet, needs to be implemented.